core/primitives/src/block.rs

```diff
@@ -333,22 +335,26 @@ impl Block {
         min_gas_price: Balance,
         max_gas_price: Balance,
     ) -> Balance {
+        // If block was skipped, the price does not change.
         if gas_limit == 0 {
-            prev_gas_price
-        } else {
-            let numerator = 2 * *gas_price_adjustment_rate.denom() as u128 * u128::from(gas_limit)
-                - *gas_price_adjustment_rate.numer() as u128 * u128::from(gas_limit)
-                + 2 * *gas_price_adjustment_rate.numer() as u128 * u128::from(gas_used);
-            let denominator =
-                2 * *gas_price_adjustment_rate.denom() as u128 * u128::from(gas_limit);
-            let new_gas_price =
-                U256::from(prev_gas_price) * U256::from(numerator) / U256::from(denominator);
-            if new_gas_price > U256::from(max_gas_price) {
-                max_gas_price
-            } else {
-                max(new_gas_price.as_u128(), min_gas_price)
-            }
+            return prev_gas_price;
         }
+
+        let gas_used = u128::from(gas_used);
+        let gas_limit = u128::from(gas_limit);
+        let adjustment_rate_numer = *gas_price_adjustment_rate.numer() as u128;
+        let adjustment_rate_denom = *gas_price_adjustment_rate.denom() as u128;
+
+        // This number can never be negative as long as gas_used <= gas_limit and
+        // adjustment_rate_numer <= adjustment_rate_denom.
+        let numerator = 2 * adjustment_rate_denom * gas_limit
```

Comment:
Would breaking down the computations into smaller chunks make it easier to understand?  E.g. 
```
let foo = 2 * adjustment_rate_denom * gas_limit;
let bar = 2 * adjustment_rate_denom * gas_limit;
let baz = 2 * adjustment_rate_denom * gas_limit;
assert(foo + bar > baz);
let numerator = foo + bar - baz;
```

I'm just not sure what are some good names for the terms above.

Reply:
Unfortunately there are no clear names for these intermediate results, they just arise due to the fact that we work with fractions, otherwise it would have been a simple linear formula as in a function comment.

I thought about directly using Rationals in the computation but that is both prone to yield another result due to rounding and we also don't have a proper-sized type to represent them (the `num-rational` crate only supports integer types up to `i64` and we need `i256` here)